cms/djangoapps/contentstore/views/course.py

```diff
@@ -1126,18 +1111,22 @@ def advanced_settings_handler(request, course_key_string):
                 return JsonResponse(CourseMetadata.fetch(course_module))
             else:
                 try:
-                    # do not process tabs unless they were modified according to course metadata
-                    filter_tabs = not _modify_tabs_to_components(request, course_module)
-
-                    # validate data formats and update
+                    # validate data formats and update the course module.
+                    # Note: don't update mongo yet, but wait until after any tabs are changed
                     is_valid, errors, updated_data = CourseMetadata.validate_and_update_from_json(
                         course_module,
                         request.json,
-                        filter_tabs=filter_tabs,
                         user=request.user,
+                        update=False,
```

Comment:
Agreed. I didn't like the ``update`` keyword either, but couldn't think of anything that seemed clearer. Breaking the method into simpler pieces with clearer names seems like the best approach.